dozer-ingestion/src/connectors/postgres/helper.rs

```diff
@@ -74,7 +74,40 @@ pub fn postgres_type_to_field(
                     .unwrap();
                     Ok(Field::from(date))
                 }
-                Type::JSONB | Type::JSON => Ok(Field::Bson(v.to_vec())),
+                Type::JSONB | Type::JSON => {
+                    let val: serde_json::Value = serde_json::from_slice(v).map_err(|_| {
+                        PostgresSchemaError::JSONBParseError(format!(
+                            "Error converting to a single row for: {}",
+                            column_type.name()
+                        ))
+                    })?;
+                    let json: JsonValue = serde_json_to_json_value(val).map_err(|_| {
+                        PostgresSchemaError::ColumnTypeNotSupported(column_type.name().to_string())
+                    })?;
+                    Ok(Field::Json(json))
+                }
+                Type::JSONB_ARRAY | Type::JSON_ARRAY => {
+                    let val: Vec<serde_json::Value> = serde_json::from_slice(v).map_err(|_| {
+                        PostgresSchemaError::JSONBParseError(format!(
+                            "Error converting to a single row for: {}",
+                            column_type.name()
+                        ))
+                    })?;
+                    let json: JsonValue = JsonValue::Array(
+                        val.into_iter()
+                            .map(|v| {
+                                serde_json_to_json_value(v)
+                                    .map_err(|_| {
+                                        PostgresSchemaError::ColumnTypeNotSupported(
```

Comment:
This is not ColumnTypeNotSupported, maybe we can forward the TypeError?

Reply:
changed